### PR TITLE
Update to Python 3.10.0

### PR DESCRIPTION
**Pull request by rco-ableton**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -7,20 +7,32 @@ FROM groovy:jre8
...
+# Add deadsnakes and install Python 3.10
+RUN add-apt-repository ppa:deadsnakes/ppa
+RUN apt-get update \
+    && apt-get install -y python3.10=3.10.* python3.10-distutils=3.10.* --no-install-recommends \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install pip
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+RUN curl --location https://bootstrap.pypa.io/get-pip.py | python3.10
```
Comment (ala-ableton) [inline Dockerfile:25]: Why do we need to install pip?

File: Dockerfile [hunk 2]
```diff
@@ -7,20 +7,32 @@ FROM groovy:jre8
...
 RUN apt-get update \
-    && apt-get install -y python3.8=3.8.* python3-pip=20.0.* --no-install-recommends \
+    && apt-get install -y software-properties-common=0.99.* --no-install-recommends \
     && apt-get clean \
     && rm -rf /var/lib/apt/lists/*
 
+# Add deadsnakes and install Python 3.10
+RUN add-apt-repository ppa:deadsnakes/ppa
+RUN apt-get update \
+    && apt-get install -y python3.10=3.10.* python3.10-distutils=3.10.* --no-install-recommends \
```
Comment (ala-ableton) [inline Dockerfile:19]: Do we really need `python3.10-distutils`?

File: Dockerfile (hunk 1 above)
Comment (rco-ableton) [reply Dockerfile:25]: It's not included in `python3.10`, so `pip install -r requirements-dev.txt` will fail.

File: Dockerfile (hunk 2 above)
Comment (rco-ableton) [reply Dockerfile:19]: If we didn't, we wouldn't, but as far as I can tell we do, so we do 😄

File: Dockerfile (hunk 1 above)
Comment (ala-ableton) [reply Dockerfile:25]: I don't see `pip install` in the Dockerfile.

Comment (rco-ableton) [reply Dockerfile:25]: You're correct, I'm mixing up `requirements.txt` and `requirements-dev.txt`. Let me check if it's really needed.

Comment (nre-ableton) [reply Dockerfile]: Seems to have worked! :fire:

Comment (rco-ableton) [reply Dockerfile]: It isn't, see https://github.com/Ableton/groovylint/pull/246/commits/4e7a8aff86dfdce40eeb7bf5069f342256f61b15. I'll squash that into https://github.com/Ableton/groovylint/pull/246/commits/40cf5ea3353e594c9acd6cfa908e53ecccae1ba4 before merging.

**Review by ala-ableton — APPROVED** [review]
Once rebased/squashed, LGTM 👍🏻

**Pull request merged**